feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go

```diff
@@ -0,0 +1,619 @@
+		{
+			desc:         "Primary connection",
+			lowID:        *ygot.Uint64(100),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectWrite:  true,
+			expectRead:   true,
+			expectStatus: 0,
+		}, {
+			desc:         "Secondary connection",
+			lowID:        *ygot.Uint64(90),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectWrite:  false,
+			expectRead:   true,
+			expectStatus: 6,
+		},
+	}
+	var p4Clients []*p4rt_client.P4RTClient
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			resp, err := streamP4RTArb(&test)
+			if err != nil && test.expectPass {
+				t.Errorf("Failed to setup P4RT Client: %v", err)
+			}
+			// Validate status code
+			if resp != test.expectStatus {
+				t.Errorf("Incorrect status code received: want %d, got %d", test.expectStatus, resp)
+			}
+			// Validate the response for Read/Write for each client
+			validateRWResp(t, &test)
+			// Keep track of connections for teardown
+			p4Clients = append(p4Clients, test.handle)
+		})
+	}
+	// Teardown clients
+	for _, p4Client := range p4Clients {
+		removeClient(p4Client)
+	}
+}
+
+// Test Duplicate ElectionID
+func TestDuplicateElectionID(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	configureDeviceId(t, dut)
+	configurePortId(t, dut)
+	testCases := []testArgs{
+		{
+			desc:         "Primary connection",
+			lowID:        *ygot.Uint64(100),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectStatus: 0,
+		}, {
+			desc:         "Secondary connection",
+			lowID:        *ygot.Uint64(100),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   false,
+			expectStatus: 0,
+		},
+	}
+	var p4Clients []*p4rt_client.P4RTClient
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			resp, err := streamP4RTArb(&test)
+			if test.expectPass {
+				if err != nil {
+					t.Errorf("Failed to setup P4RT Client: %v", err)
+				}
+			} else {
+				if err != nil {
+					t.Logf("As expected failed to setup P4RT Client: %v", err)
+				}
+			}
+
+			// Validate status code
+			if resp != test.expectStatus {
+				t.Errorf("Incorrect status code received: want %d, got %d", test.expectStatus, resp)
+			}
+			// Keep track of connections for teardown
+			p4Clients = append(p4Clients, test.handle)
+		})
+	}
+	// Teardown clients
+	for _, p4Client := range p4Clients {
+		removeClient(p4Client)
+	}
+}
+
+func TestReplacePrimary(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	configureDeviceId(t, dut)
+	configurePortId(t, dut)
+	testCases := []mcTestArgs{
+		{
+			name:       "Primary_secondary_OK",
+			expectPass: true,
+			Items: [2]*testArgs{
+				{
+					desc:              "Primary connection",
```

Comment:
The only variable I see are the "desc" and the "lowID". Can you just use the table for those and create a single testArg